### DISCUSSION TITLE
Having trouble with date-time validation

### DISCUSSION THREAD

**Discussion by bbengfort** [rajnandan1/frogmentwebsite#2] [General]
Hello,

First - thank you so much for making Frogment; we have a fairly large OpenAPI spec (open source, available here: https://trisa.dev/envoy/openapi/index.html) and managing it as a single yaml document has been a bit of a nightmare. We really like being able to use Frogment to break things into discrete chunks and organize our work. 

I'm currently using the Mac Desktop App version of Frogment, and I've been having a couple of issues with the validator. I was wondering if there was a fix or a configuration I should change to make it work a bit better for me. Right now I'm getting the warning:

```
oas3-valid-schema-example: jsonschema: invalid jsonType: time.Time
```

For the following example:

```
example: 2024-08-28T12:23:24-05:00
```

The date-time is correctly formatted as an RFC 3339 timestamp, but the warning seems to be a javascript issue rather than a validation error. Is my timestamp example incorrect? If not, is there a way I can fix the jsonschema to correctly validate it? Baring that is there anyway to ignore specific warnings and informs using Frogment? 

Also, I've been having issues when the full spec is fully valid (e.g. no errors, warnings, or informs); I've been getting the following error:

```
TypeError: undefined is not an object (evaluating 'n.slice')
```

Any help or guidance would be appreciated. Thank you so much for your hard work on this project and your open source contributions.

Best,
Benjamin Bengfort

**Comment by bbengfort** [comment]
Note that I've also tried the following examples, and received the following warnings:

- `example: 2024-08-28T12:23:24Z`: `oas3-valid-schema-example: jsonschema: invalid jsonType: time.Time`
- `example: 2024-08-28T10:14:43+00:00`: `oas3-valid-schema-example: jsonschema: invalid jsonType: time.Time`
- `example: 2024-08-28T10:14:43`: `oas3-valid-schema-example: '2024-08-28T10:14:43' is not valid 'date-time'`
- `example: 2024-08-28 10:14:43`: `oas3-valid-schema-example: jsonschema: invalid jsonType: time.Time`

**Reply by rajnandan1** [reply]
Hi Benjamin

I just tried the same spec, I don't see any errors, Can you provide screenshot or exact schema name where you are facing the above issues?

**Reply by bbengfort** [reply]
When I use the website, I'm not getting any warnings; it's just when I'm using the Mac Desktop App. Here is the screenshot:

![Screenshot 2024-09-01 at 12 06 05](https://github.com/user-attachments/assets/860bd895-b068-46f0-8c69-f6462d35f61c)

All 52 of the warnings are the same invalid jsonType warning.
